build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -0,0 +1,249 @@
+    this.client.amp = {
+      useCompiledJs: !!argv.compiled,
+      saucelabs: !!argv.saucelabs,
+      singlePass: !!argv.single_pass,
+      adTypes: getAdTypes(),
+      mochaTimeout: this.client.mocha.timeout,
+      propertiesObfuscated: !!argv.single_pass,
+      testServerPort: this.client.testServerPort,
+      testOnIe:
+        this.browsers.includes('IE') || this.browsers.includes('SL_IE_11'),
+    };
+
+    if (argv.coverage && this.testType != 'a4a') {
+      this.plugins.push('karma-coverage-istanbul-reporter');
+      this.coverageIstanbulReporter = {
+        dir: 'test/coverage',
+        reports: isTravisBuild()
+          ? ['lcovonly']
+          : ['html', 'text', 'text-summary'],
+        'report-config': {lcovonly: {file: `lcov-${testType}.info`}},
+      };
+
+      const plugin = [
+        'istanbul',
+        {
+          exclude: [
+            'ads/**/*.js',
+            'third_party/**/*.js',
+            'test/**/*.js',
+            'extensions/**/test/**/*.js',
+            'testing/**/*.js',
+          ],
+        },
+      ];
+
+      this.browserify.transform = [['babelify', {plugins: [plugin]}]];
+    }
+  }
+
+  getBrowsersObject() {
+    if (argv.saucelabs) {
+      if (this.testType == 'unit') {
+        return {browsers: ['SL_Safari_12', 'SL_Firefox']};
+      }
+
+      if (this.testType == 'integration') {
+        return {
+          browsers: [
+            'SL_Chrome',
+            'SL_Firefox',
+            'SL_Edge_17',
+            'SL_Safari_12',
+            'SL_IE_11',
+            // TODO(amp-infra): Evaluate and add more platforms here.
+            //'SL_Chrome_Android_7',
+            //'SL_iOS_11',
+            //'SL_iOS_12',
+            'SL_Chrome_Beta',
+            'SL_Firefox_Beta',
+          ],
+        };
+      }
+
+      throw new Error(
+        'The --saucelabs flag is valid only for `gulp unit` and `gulp integration`.'
+      );
+    }
+
+    const chromeFlags = [];
+    if (argv.chrome_flags) {
+      argv.chrome_flags.split(',').forEach(flag => {
+        chromeFlags.push('--'.concat(flag));
+      });
+    }
+
+    const options = new Map();
+    options
+      .set('chrome_canary', {browsers: ['ChromeCanary']})
+      .set('chrome_flags', {
+        browsers: ['Chrome_flags'],
+        customLaunchers: {
+          // eslint-disable-next-line
+          Chrome_flags: {
+            base: 'Chrome',
+            flags: chromeFlags,
+          },
+        },
+      })
+      .set('edge', {browsers: ['Edge']})
+      .set('firefox', {browsers: ['Firefox']})
+      .set('headless', {browsers: ['Chrome_no_extensions_headless']})
+      .set('ie', {
+        browsers: ['IE'],
+        customLaunchers: {
+          IeNoAddOns: {
+            base: 'IE',
+            flags: ['-extoff'],
+          },
+        },
+      })
+      .set('safari', {browsers: ['Safari']});
+
+    for (const key of Array.from(options.keys())) {
```

Comment:
do you need the extra `Array.from` call here? `for of` should handle the `Iterable`  that `keys()` returns

Reply:
Good catch, I think I used Array because I wanted to exit the loop early and do something else, but here I changed it to return instead so I don't need it anymore.